pkg/lso/localvolumeset_test.go

```diff
@@ -0,0 +1,612 @@
+			expectedStatus:     false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		exist := testCase.testLocalVolumeSet.Exists()
+		assert.Equal(t, testCase.expectedStatus, exist)
+	}
+}
+
+func TestLocalVolumeSetGet(t *testing.T) {
+	testCases := []struct {
+		testLocalVolumeSet *LocalVolumeSetBuilder
+		expectedError      error
+	}{
+		{
+			testLocalVolumeSet: buildValidLocalVolumeSetObjectBuilder(buildLocalVolumeSetClientWithDummyObject()),
+			expectedError:      nil,
+		},
+		{
+			testLocalVolumeSet: buildInValidLocalVolumeSetObjectBuilder(buildLocalVolumeSetClientWithDummyObject()),
+			expectedError:      fmt.Errorf("localVolumeSet 'name' cannot be empty"),
+		},
+		{
+			testLocalVolumeSet: buildValidLocalVolumeSetObjectBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError:      fmt.Errorf("localvolumesets.local.storage.openshift.io \"ocs-deviceset\" not found"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		localVolumeSetObj, err := testCase.testLocalVolumeSet.Get()
+
+		if testCase.expectedError == nil {
+			assert.Equal(t, localVolumeSetObj.Name, testCase.testLocalVolumeSet.Definition.Name)
+			assert.Equal(t, localVolumeSetObj.Namespace, testCase.testLocalVolumeSet.Definition.Namespace)
+			assert.Nil(t, err)
+		} else {
+			assert.Equal(t, testCase.expectedError.Error(), err.Error())
+		}
+	}
+}
+
+func TestLocalVolumeSetCreate(t *testing.T) {
+	testCases := []struct {
+		testLocalVolumeSet *LocalVolumeSetBuilder
+		expectedError      string
+	}{
+		{
+			testLocalVolumeSet: buildValidLocalVolumeSetObjectBuilder(buildLocalVolumeSetClientWithDummyObject()),
+			expectedError:      "",
+		},
+		{
+			testLocalVolumeSet: buildInValidLocalVolumeSetObjectBuilder(buildLocalVolumeSetClientWithDummyObject()),
+			expectedError:      "localVolumeSet 'name' cannot be empty",
+		},
+		{
+			testLocalVolumeSet: buildValidLocalVolumeSetObjectBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError:      "",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testLocalVolumeSetBuilder, err := testCase.testLocalVolumeSet.Create()
+
+		if testCase.expectedError == "" {
+			assert.Equal(t, testLocalVolumeSetBuilder.Definition.Name, testLocalVolumeSetBuilder.Object.Name)
+			assert.Equal(t, testLocalVolumeSetBuilder.Definition.Namespace, testLocalVolumeSetBuilder.Object.Namespace)
+			assert.Nil(t, err)
+		} else {
+			assert.Equal(t, testCase.expectedError, err.Error())
+		}
+	}
+}
+
+func TestLocalVolumeSetDelete(t *testing.T) {
+	testCases := []struct {
+		testLocalVolumeSet *LocalVolumeSetBuilder
+		expectedError      error
+	}{
+		{
+			testLocalVolumeSet: buildValidLocalVolumeSetObjectBuilder(buildLocalVolumeSetClientWithDummyObject()),
+			expectedError:      nil,
+		},
+		{
+			testLocalVolumeSet: buildValidLocalVolumeSetObjectBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError:      fmt.Errorf("localVolumeSet cannot be deleted because it does not exist"),
```

Comment:
Please update Delete function.
We don't return error if object doesn't exist. Just print log message and return nil. 
Please refer to:
https://github.com/openshift-kni/eco-goinfra/blob/82c1862d9c98d00fac446c4bf84d37ca5ac36e08/pkg/argocd/applications.go#L162